### PR TITLE
list plugin

### PR DESCRIPTION
**Pull request by edeustace**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,9 +7,7 @@
     "lint": "./node_modules/.bin/eslint --ext .js --ext .jsx packages"
   },
   "devDependencies": {
-    "lerna": "^2.8.0",
```
Comment (edeustace) [inline package.json]: don't remove `babel-eslint` and `es-lint-plugin-react` from the root `package.json` - these are configured at the root of this mono-repo, so the individual packages don't need to do it. Also don't add `babel-eslint` and `lerna` to the package's `package.json`.

File: packages/editable-html/src/plugins/list/__test__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import React from "react";
+
+import {configure} from "enzyme";
+import Adapter from "enzyme-adapter-react-15";
+import renderer from 'react-test-renderer';
+import List from './../../list/index';
```
Comment (edeustace) [inline packages/editable-html/src/plugins/list/__test__/index.test.js]: `import List, {serialization} from '..''`

File: packages/editable-html/src/plugins/list/__test__/index.test.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+import Util from '../../utils';
+
+configure({adapter : new Adapter() });
+
+describe("ListPlugin", () => {
+  let list, ListPlugin, toolbar;
+  beforeEach(() => {
+    list = require('../index');
```
Comment (edeustace) [inline packages/editable-html/src/plugins/list/__test__/index.test.js]: this is unnecessary - you've already imported `List` and co above.

File: packages/editable-html/src/plugins/list/__test__/index.test.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    ListPlugin = list.default;
+    toolbar = list.toolbar
+  });
+
+  describe("deserialize", () => {
+    it("should deserialize list item", () => {
```
Comment (edeustace) [inline packages/editable-html/src/plugins/list/__test__/index.test.js]: There's a fair bit of boilerplate in the following tests, could we add a shared function?

```javascript
const assertDeserialize = (tagName, expectedType) => {
  it(`deserializes: ${tagName} to ${expectedType}`, () = > {
    const out = serialization.deserialize({tagName}, jest.fn());
    expect(out).toEqual({object: 'block', type:expectedType, nodes: undefined});
  });
}

assertDeserialization('li', 'list_item');
assertDeserialization('ul', 'ul_list');
assertDeserialization('ol', 'ol_list');

```

File: packages/editable-html/src/plugins/list/__test__/index.test.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        childNodes : []
+      }
+      const next = jest.fn();
+      const out = list.serialization.deserialize(el, next);
+
+      expect(out).toMatchObject({object : 'block', type : 'ol_list', nodes: undefined});
+    });        
+  });
+
+  describe("serialize", () => {
```
Comment (edeustace) [inline packages/editable-html/src/plugins/list/__test__/index.test.js]: As above - lot of boilerplate - can we shrink it w/ a shared function?

File: packages/editable-html/src/plugins/list/__test__/index.test.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        type:'div',
+        text : 'hello'
+      });
+
+      expect(out.type).toMatch('ol');
+    });        
+  });
+  describe("renderNode", () => {
+    it("should render list item node", () => {
+      let obj = {
```
Comment (edeustace) [inline packages/editable-html/src/plugins/list/__test__/index.test.js]: As above - lot of boiler plate - can we shrink w/ shared node?

File: packages/editable-html/src/plugins/list/__test__/index.test.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+      expect(out.type).toMatch('ol');
+    });    
+  });  
+
+  describe("toolbar", () => {
+    describe('onClick toolbar ul', () => {
+     
+     it("should onClick ul", () => {
+      
+      let plugin = ListPlugin({});
```
Comment (edeustace) [inline packages/editable-html/src/plugins/list/__test__/index.test.js]: empty test?
and missing test for `isActive`?

File: packages/editable-html/src/plugins/list/__test__/index.test.js (hunk 3 above)
Comment (mukeshsharma04) [reply packages/editable-html/src/plugins/list/__test__/index.test.js]: Removed

File: packages/editable-html/src/plugins/list/__test__/index.test.js (hunk 2 above)
Comment (mukeshsharma04) [reply packages/editable-html/src/plugins/list/__test__/index.test.js]: Done

File: packages/editable-html/src/plugins/list/__test__/index.test.js (hunk 4 above)
Comment (mukeshsharma04) [reply packages/editable-html/src/plugins/list/__test__/index.test.js]: Done

File: packages/editable-html/src/plugins/list/__test__/index.test.js (hunk 5 above)
Comment (mukeshsharma04) [reply packages/editable-html/src/plugins/list/__test__/index.test.js]: Done

File: packages/editable-html/src/plugins/list/__test__/index.test.js (hunk 6 above)
Comment (mukeshsharma04) [reply packages/editable-html/src/plugins/list/__test__/index.test.js]: Done

File: package.json (hunk 1 above)
Comment (mukeshsharma04) [reply package.json]: ok

File: packages/editable-html/src/plugins/list/__test__/index.test.js (hunk 7 above)
Comment (mukeshsharma04) [reply packages/editable-html/src/plugins/list/__test__/index.test.js]: need your help to understand to write same.

**Pull request merged**